admin/coupon_admin.php

```diff
@@ -309,12 +309,12 @@
     ];
 
     zen_db_perform(TABLE_COUPONS, $sql_data_array);
-    $cid = $db->insert_ID();
+    $cid = $db->insert_ID(); // Note this may be a string if the newly added value exceeds limit of an integer. Thus later conversion back to an integer may be problematic.
```

Comment:
The addition of this comment suggests that a site is expected to have more than 2,147,483,647 (yes, that's 2 billion) coupons.

(2147483647 is MySQL's max unsigned `integer` value, and also PHP's 32-bit max value (64-bit is way higher))

Reply:
Again, just carrying over @mc12345678's PR and applying my requested changes.  I'll remove the comment.